stable_plugins/visualization/file_types/qasm_visualization.py

```diff
@@ -0,0 +1,252 @@
+            name=plugin.name,
+            version=plugin.version,
+            type=PluginType.visualization,
+            entry_point=EntryPoint(
+                href=url_for(f"{QASM_BLP.name}.ProcessView"),
+                ui_href=url_for(f"{QASM_BLP.name}.MicroFrontend"),
+                plugin_dependencies=[],
+                data_input=[
+                    InputDataMetadata(
+                        data_type="*",
+                        content_type=["text/x-qasm"],
+                        parameter="data",
+                        required=True,
+                    )
+                ],
+                data_output=[
+                    DataMetadata(
+                        data_type="*",
+                        content_type=["image/svg+xml"],
+                        required=True,
+                    )
+                ],
+            ),
+            tags=[],
+        )
+
+
+@QASM_BLP.route("/ui/")
+class MicroFrontend(MethodView):
+    """Micro frontend for the QASM visualization plugin."""
+
+    @QASM_BLP.html_response(
+        HTTPStatus.OK, description="Micro frontend of the QASM visualization plugin."
+    )
+    @QASM_BLP.arguments(
+        QasmInputParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="query",
+        required=False,
+    )
+    @QASM_BLP.require_jwt("jwt", optional=True)
+    def get(self, errors):
+        """Return the micro frontend."""
+        return self.render(request.args, errors, False)
+
+    @QASM_BLP.html_response(
+        HTTPStatus.OK, description="Micro frontend of the QASM visualization plugin."
+    )
+    @QASM_BLP.arguments(
+        QasmInputParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="form",
+        required=False,
+    )
+    @QASM_BLP.require_jwt("jwt", optional=True)
+    def post(self, errors):
+        """Return the micro frontend with prerendered inputs."""
+        return self.render(request.form, errors, not errors)
+
+    def render(self, data: Mapping, errors: dict, valid: bool):
+        plugin = QasmVisualization.instance
+        if plugin is None:
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR)
+        schema = QasmInputParametersSchema()
+        return Response(
+            render_template(
+                "qasm_visualization.html",
+                name=plugin.name,
+                version=plugin.version,
+                schema=schema,
+                valid=valid,
+                values=data,
+                errors=errors,
+                example_values=url_for(f"{QASM_BLP.name}.MicroFrontend"),
+            )
+        )
+
+
+@QASM_BLP.route("/circuits/")
+def get_circuit_image():
+    circuit_url = request.args.get("circuit-url", None)
+    if circuit_url is None:
+        abort(HTTPStatus.NOT_FOUND)
+    filename = hashlib.md5(circuit_url.encode("utf-8")).hexdigest() + ".png"
```

Comment:
I changed the hash to `sha256`. Currently, only the hash of the file URL is hashed. Does it make more sense to hash the file contents to distinguish different circuits?

Reply:
not really, in QHAna every file version gets its own unique URL. For outside data it should be enough to clear cache entries that are older than x hours or days and regenerate them instead. (This would require storing a time...)